iceoryx_posh/include/iceoryx_posh/internal/popo/ports/client_server_port_types.hpp

```diff
@@ -0,0 +1,160 @@
+#include "iceoryx_posh/internal/popo/building_blocks/locking_policy.hpp"
+#include "iceoryx_utils/internal/relocatable_pointer/relative_ptr.hpp"
+
+#include <cstdint>
+
+namespace iox
+{
+namespace popo
+{
+struct ClientChunkDistributorConfig
+{
+    static constexpr uint32_t MAX_QUEUES = 1;
+    static constexpr uint64_t MAX_HISTORY_CAPACITY = 1; // could be 0, but problem for the container then
+};
+
+struct ServerChunkDistributorConfig
+{
+    static constexpr uint32_t MAX_QUEUES = MAX_CLIENTS_PER_SERVER;
+    static constexpr uint64_t MAX_HISTORY_CAPACITY = 1; // could be 0, but problem for the container then
+};
+
+struct ClientChunkQueueConfig
+{
+    static constexpr uint64_t MAX_QUEUE_CAPACITY = MAX_RESPONSE_QUEUE_CAPACITY;
+};
+
+struct ServerChunkQueueConfig
+{
+    static constexpr uint64_t MAX_QUEUE_CAPACITY = MAX_REQUEST_QUEUE_CAPACITY;
+};
+
+using ClientChunkQueueData_t = ChunkQueueData<ClientChunkQueueConfig, ThreadSafePolicy>;
+
+using ServerChunkQueueData_t = ChunkQueueData<ServerChunkQueueConfig, ThreadSafePolicy>;
+
+using ClientChunkDistributorData_t =
+    ChunkDistributorData<ClientChunkDistributorConfig, ThreadSafePolicy, ChunkQueuePusher<ServerChunkQueueData_t>>;
+
+using ServerChunkDistributorData_t =
+    ChunkDistributorData<ServerChunkDistributorConfig, ThreadSafePolicy, ChunkQueuePusher<ClientChunkQueueData_t>>;
+
+using ClientChunkReceiverData_t = ChunkReceiverData<MAX_RESPONSES_PROCESSED_SIMULTANEOUSLY, ClientChunkQueueData_t>;
+
+using ServerChunkReceiverData_t = ChunkReceiverData<MAX_REQUESTS_PROCESSED_SIMULTANEOUSLY, ServerChunkQueueData_t>;
+
+using ClientChunkSenderData_t = ChunkSenderData<MAX_REQUESTS_ALLOCATED_SIMULTANEOUSLY, ClientChunkDistributorData_t>;
+
+using ServerChunkSenderData_t = ChunkSenderData<MAX_RESPONSES_ALLOCATED_SIMULTANEOUSLY, ServerChunkDistributorData_t>;
+
+class RequestHeader
+{
+  public:
+    explicit RequestHeader(cxx::not_null<ClientChunkQueueData_t* const> chunkQueueDataPtr) noexcept
+        : m_responseQueue(chunkQueueDataPtr)
+    {
+    }
+
+    RequestHeader(const RequestHeader& other) = delete;
+    RequestHeader& operator=(const RequestHeader&) = delete;
+    RequestHeader(RequestHeader&& rhs) = default;
+    RequestHeader& operator=(RequestHeader&& rhs) = default;
+    virtual ~RequestHeader() = default;
+
+    void setSequenceNumber(int64_t sequenceNumber) noexcept
+    {
+        m_sequenceNumber = sequenceNumber;
+    }
+    void setFireAndForget(bool fireAndForget) noexcept
+    {
+        m_isFireAndForget = fireAndForget;
+    }
+
+    mepoo::ChunkHeader* getChunkHeader() noexcept
+    {
+        /// todo
+        return nullptr;
+    }
+    void* getPayload() noexcept
+    {
+        /// todo
+        return nullptr;
+    }
+
+  private:
+    int64_t m_sequenceNumber{0};
+    bool m_isFireAndForget{false};
+    relative_ptr<ClientChunkQueueData_t> m_responseQueue;
+};
+
+class ResponseHeader
```

Comment:
will do when I implement things. This was just the API so far

Reply:
Well, it's not a big deal to create a new file, isn't it? ;-) But no hard feelings on this, but IMHO it would be cleaner start.

Reply:
Sure. My intention was also to get feedback on the general interface. Maybe it would have turned out that the request and response headers are no good idea. So I wanted to save effort until it is consolidated